docs/proposals/kubernetes-registry.md

```diff
@@ -0,0 +1,210 @@
+# Kubernetes Registry Implementation for ToolHive Operator
+
+## Problem Statement
+
+The ToolHive operator currently supports managing individual MCP servers through `MCPServer` CRDs, but lacks a centralized registry mechanism within Kubernetes. This creates challenges in discoverability, catalog management, upstream compatibility, and operational complexity.
+
+## Goals
+
+- **Native Kubernetes Registry**: Implement registry functionality using Custom Resource Definitions
+- **Upstream Format Support**: Leverage existing upstream conversion capabilities for ecosystem compatibility
+- **Multi-Registry Support**: Support both local registry entries and external registry synchronization. 
+- **Registry Hierarchy**: Support the multi-registry hierarchy defined in the upstream model
+- **Application Integration**: Provide REST API for programmatic access to registry data
+- **GitOps Compatibility**: Enable declarative registry management through CRD-based operations
+
+## Architecture Overview
+
+The Kubernetes registry implementation extends the operator with the `MCPRegistry` CRD and supporting controllers that work with the existing MCPServer CRD to provide a complete registry-to-deployment workflow.
+
+## CRD Design Overview
+
+### MCPRegistry CRD
+
+The `MCPRegistry` CRD represents a registry source and synchronization configuration with these key components:
+
+- **Source Configuration**: Support for ConfigMap, URL, Git, and Registry API sources
+- **Format Specification**: Handle both ToolHive and upstream registry formats  
+- **Sync Policy**: Automatic and manual synchronization with configurable intervals
+- **Filtering**: Include/exclude servers based on names, tags, tiers, and transports
+
+### Job CRDs (Phase 3)
+
+Declarative operation CRDs for GitOps compatibility:
+- `MCPRegistryImportJob`: Declarative import operations
+- `MCPRegistryExportJob`: Declarative export operations  
```

Comment:
What would be the use-case for export?

Reply:
Data backup? Anyway, I agree it can be dropped for now: if the original data source is not mutable from the registry itself (we only import and sync), then it's probably useless.